### PR TITLE
Flow tests with page objects

### PR DESCRIPTION
**Pull request by laeschjs**
Closes #X

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -235,8 +235,8 @@ PLATFORMS
   ruby
 
 DEPENDENCIES
-  ae_page_objects
   acts-as-taggable-on (~> 6.0)
+  ae_page_objects
```
Comment (lsq921224) [inline Gemfile.lock]: this change should be in another commit, because it's not part of `added image destroy action`

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -21,6 +21,15 @@ def show
     redirect_to action: 'new'
   end
 
+  def destroy
+    Image.destroy(params[:id])
+    flash[:success] = 'Image Deleted'
+  rescue StandardError
```
Comment (lsq921224) [inline app/controllers/images_controller.rb]: `StandardError` is too broad, here we are only handling `ActiveRecord::RecordNotFound`

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -43,15 +48,28 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+  test 'should delete' do
+    delete image_path(@image)
+
+    assert_redirected_to root_path
+    follow_redirect!
+    assert_select '#flash_success_message', 'Image Deleted'
+  end
+
+  test 'should not delete with bad input' do
+    delete image_path(id: 1000)
```
Comment (lsq921224) [inline test/controllers/images_controller_test.rb]: I recommend using id 0 or -1, because a positive number like 1000 may be legit in some case. For example, you run this test file 1000 times

File: app/views/application/home.html.erb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <a href=<%= new_image_path %>>Upload</a> an Image!
 <br />
-
+<div id="flash_success_message"><%= flash[:success] %></div>
+<div id="flash_error_message"><%= flash[:error] %></div>
```
Comment (lsq921224) [inline app/views/application/home.html.erb]: These two lines should be in previous commit because you were already using them

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -51,31 +55,85 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
 
-    assert_redirected_to root_path
+    assert_redirected_to images_path
     follow_redirect!
     assert_select '#flash_error_message', 'Image not found'
   end
+
+  test 'index page allows the user to filter by tag' do
+    get images_path
+    assert_response :success
```
Comment (lsq921224) [inline test/controllers/images_controller_test.rb]: `assert_response :ok` is preferred over `:success` because every `2xx` will pass `:success`, only `200` will pass `:ok`

File: test/page_objects/images/index_page.rb [hunk 6]
```diff
@@ -3,23 +3,37 @@ module Images
...
       def add_new_image!
-        node.click_on('New Image')
+        node.click_on('Upload an Image')
+        stale!
         window.change_to(NewPage)
       end
 
       def showing_image?(url:, tags: nil)
-        # TODO
+        images.each do |image|
```
Comment (lsq921224) [inline test/page_objects/images/index_page.rb]: There are cleaner ways to write this method with `Array#any?` method. Can you try using it?

File: app/controllers/images_controller.rb (hunk 2 above)
Comment (laeschjs) [reply app/controllers/images_controller.rb]: fixed

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (laeschjs) [reply test/controllers/images_controller_test.rb]: fixed

File: test/controllers/images_controller_test.rb (hunk 5 above)
Comment (laeschjs) [reply test/controllers/images_controller_test.rb]: fixed

File: test/page_objects/images/index_page.rb (hunk 6 above)
Comment (laeschjs) [reply test/page_objects/images/index_page.rb]: fixed

File: app/views/application/home.html.erb (hunk 4 above)
Comment (laeschjs) [reply app/views/application/home.html.erb]: fixed

File: Gemfile.lock (hunk 1 above)
Comment (laeschjs) [reply Gemfile.lock]: fixed
